public/app/api/createBaseQuery.ts

```diff
@@ -34,10 +38,10 @@ export function createBaseQuery({ baseURL }: { baseURL: string }): BaseQueryFn<R
 
 export function handleRequestError(error: unknown) {
   if (isFetchError(error)) {
-    return { error: new Error(error.data.message) };
-  } else if (error instanceof Error) {
     return { error };
+  } else if (error instanceof Error) {
+    return { error: { message: error } };
   } else {
-    return { error: new Error('Unknown error') };
+    return { error: { message: new Error('Unknown error') } };
```

Comment:
I think this should just be

```
  } else if (error instanceof Error) {
    return { error };
  } else {
    return { error: new Error('Unknown error') };
```

Returns the error directly on the error property, you can consistently narrow it down, and in avoids `resp.error.message.message`.